modules/util/portal-tools-service-builder/src/main/java/com/liferay/portal/tools/service/builder/ServiceBuilder.java

```diff
@@ -5879,34 +5879,37 @@ private boolean _hasFinderThatIsNotUniqueByCompany(
 			return false;
 		}
 
-		boolean hasCompanyId = Stream.of(
-			columnElements.toArray(new Element[0])
-		).map(
-			columnElement -> columnElement.attributeValue("name")
-		).anyMatch(
-			columnName -> columnName.equals("companyId")
-		);
+		boolean hasCompanyId = false;
+
+		for (Element columnElement : columnElements) {
+			String columnName = columnElement.attributeValue("name");
+
+			if (columnName.equals("companyId")) {
+				hasCompanyId = true;
+
+				break;
+			}
+		}
 
 		if (!hasCompanyId) {
 			return false;
 		}
 
-		String[] finderColumnNames = Stream.of(
-			finderColumnElements.toArray(new Element[0])
-		).map(
-			finderColumnElement -> finderColumnElement.attributeValue("name")
-		).filter(
+		List<String> finderColumnNames = ListUtil.filter(
```

Comment:
@slnn Could you just merge the `TransformUtil.transform` and `ListUtil.filter`?

Reply:
Hi @dantewang 

What do you mean about merging the TransformUtil.transform and ListUtil.filter?

Reply:
@slnn In `TransformUtil.transform`'s mapping function, get the name attribute, perform a null check as well as the check in `ListUtil.filter`. If not pass, return null; otherwise, return the name attribute. This way, transform and filter happens in one iteration.